operator/pkg/ingress/ingress.go

```diff
@@ -0,0 +1,447 @@
+		return err
+	}
+
+	log.WithField("endpoints", key).Info("Created Endpoints for Ingress")
+	return nil
+}
+
+func (ic *IngressController) createEnvoyConfig(ingress *slim_networkingv1.Ingress) error {
+	desired, err := getEnvoyConfigForIngress(k8s.Client(), ingress)
+	if err != nil {
+		return err
+	}
+
+	// check if the CiliumEnvoyConfig resource already exists
+	key, err := cache.MetaNamespaceKeyFunc(desired)
+	if err != nil {
+		log.WithError(err).Warn("MetaNamespaceKeyFunc returned an error")
+		return err
+	}
+	existingEnvoyConfig, exists, err := ic.envoyConfigManager.getByKey(key)
+	if err != nil {
+		log.WithError(err).Warn("CiliumEnvoyConfig lookup failed")
+		return err
+	}
+	if exists {
+		if desired.DeepEqual(existingEnvoyConfig) {
+			log.WithField("cilium-envoy-config", key).Info("No change for existing CiliumEnvoyConfig")
+			return nil
+		}
+		// Update existing CEC
+		newEnvoyConfig := existingEnvoyConfig.DeepCopy()
+		newEnvoyConfig.Spec = desired.Spec
+		_, err = k8s.CiliumClient().CiliumV2alpha1().CiliumEnvoyConfigs(ingress.Namespace).Update(context.Background(), newEnvoyConfig, metav1.UpdateOptions{})
+		if err != nil {
+			log.WithError(err).WithField("ingress", ingress.Name).Error("Failed to update CiliumEnvoyConfig for ingress")
+		}
+		log.WithField("ingress", ingress.Name).Error("Updated CiliumEnvoyConfig for ingress")
+		return nil
+	}
+	_, err = k8s.CiliumClient().CiliumV2alpha1().CiliumEnvoyConfigs(ingress.Namespace).Create(context.Background(), desired, metav1.CreateOptions{})
+	if err != nil {
+		log.WithError(err).WithField("ingress", ingress.Name).Error("Failed to create CiliumEnvoyConfig for ingress")
+		return err
+	}
+	log.WithField("ingress", ingress.Name).Info("Create CiliumEnvoyConfig for ingress")
+	return nil
+}
+
+func (ic *IngressController) handleAddIngress(obj interface{}) {
+	if ingress := k8s.ObjToV1Ingress(obj); ingress != nil {
+		ic.queue.Add(ingressAddedEvent{ingress: ingress})
+	}
+}
+
+func (ic *IngressController) handleUpdateIngress(oldObj, newObj interface{}) {
+	oldIngress := k8s.ObjToV1Ingress(oldObj)
+	if oldIngress == nil {
+		return
+	}
+	newIngress := k8s.ObjToV1Ingress(newObj)
+	if newIngress == nil {
+		return
+	}
+	ic.queue.Add(ingressUpdatedEvent{oldIngress: oldIngress, newIngress: newIngress})
+}
+
+func (ic *IngressController) handleDeleteIngress(obj interface{}) {
+	if ingress := k8s.ObjToV1Ingress(obj); ingress != nil {
+		ic.queue.Add(ingressDeletedEvent{ingress: ingress})
+	}
+}
+
+func (ic *IngressController) deleteCiliumEnvoyConfig(ingress *slim_networkingv1.Ingress) error {
+	// check if the CiliumEnvoyConfig resource exists.
+	resourceName := getCECNameForIngress(ingress)
+	_, exists, err := ic.envoyConfigManager.getByKey(resourceName)
+	if err != nil {
+		log.WithError(err).Warn("CiliumEnvoyConfig lookup failed")
+		return err
+	}
+	if !exists {
+		log.WithField("cilium-envoy-config", resourceName).Info("CiliumEnvoyConfig already deleted. Continuing...")
```

Comment:
same for the Info messages. I didn't catch all of them but it seems that a lot of the Info messages could be changed to Debug instead.